study-frontend/src/Components/SignInBox.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useEffect } from 'react';
+import { Button, Checkbox, Form, Input } from 'antd';
+import Registration from '../Services/Authentication/Registration';
+import { useNavigate } from 'react-router-dom';
+import { routes } from '../constants';
+import '../cssFiles/loginSignInPage.css'
+
+const onFinishFailed = (errorInfo: any) => {
+  console.log('Failed:', errorInfo);
+};
+
+
+
+
+
+
+
+type Props = {
+    handleClikc : () => void; 
+};
+
+
+const SignIn = ({handleClikc} : Props) => {
+    const [response, setResponse] = useState();
+    const [username, setUsername] = useState("");
+    const [email, setEmail] = useState("");
+    const [password, setPassword] = useState("");
+    const [isError, setIsError] = useState(false);
+    const navigate = useNavigate(); 
+    const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+    
+    const [message, setMessage] = useState("")
+    
+    const handleSubmit = async() => {
+        if(username  !== "" && email !== "" && password !== ""){
+            try{
+                console.log(username)
+                const response = await Registration({username,email,password}); 
+                console.log(response);
+                sessionStorage.setItem("jwt", response.token); 
```

Comment:
Make sure to not repeat yourself if you endup having to copy pased the same code around it can be moved to a seperate shared place.